interpolated_model_timeseries.py

```diff
@@ -1,96 +1,81 @@
-def get_catda(wrfouts, pol, level):
+def get_catda(wrfouts, pol, level, tdim):
+    """
+    Description:
+        Returns a concatenated data array of the files contained within.
+    Args:    
+        wrfouts (list): List of wrfout files.
+        pol (str): Pollutant.
+        level (int): Model level as an index location.
+        tdim (str): Name of the time dimension.
+    Returns:
+        catda (xarray DataArray): Concatenate data array.
+    """
     da_list = []
     for filepath in wrfouts:
-        
-        ds = xr.open_dataset(filepath)
-        da = ds[pol]
-            
-        # chunk along time dim if length <1 
-        tdim_len = len(da[time])
+        with xr.open_dataset(filepath) as ds:
+            da = ds[pol]
+
+        tdim_len = len(da[tdim])
         if tdim_len > 1:
-            da = da.chunk({time:tdim_len//10})
-        
-        if 'bottom_top' in da.dims:
-            # slice by level
+            da = da.chunk({tdim:tdim_len//10})
+        elif 'bottom_top' in da.dims:
             da = da.loc[{'bottom_top':level}]
-        
+
         da_list.append(da)
-        
-        
-    catda = xr.concat(da_list, dim=time)
-    
-    return(catda)
-    
-def interpolate_model_timeseries(locations, times='all'):
-    
-    catda = get_catda(wrfouts=wrfouts, pol=pol, level=level)
-    
-    time_coord = np.arange(0, len(catda[time]))
+    catda = xr.concat(da_list, dim=tdim)
+    return catda
     
+def interpolate_model_timeseries(locations, tdim, times='all'):
+    """
+    Description:
+        Interpolates timeseries at given lat/lon locations.
+        The wrfout files must be regridded to a rectilinear grid.
+    Args:
+        locations (list of tuples): Observation lat lon locations.
+        tdim (str): Name of the time dimension.
+        times (str, optional): Times of interpolation, default = 'all'.
+    Returns:
+        df (pandas DataFrame): Interpolated timeseries.
+    """
+    catda = get_catda(wrfouts, pol, level, tdim)
+    time_coord = np.arange(0, len(catda[tdim]))
     print('creating interpolator...')
     f = RegularGridInterpolator((time_coord, 
                                  catda.coords['lat'].values,
                                  catda.coords['lon'].values),
                                  catda.values
                                 )
     print('...done')
-    
-    df = pd.DataFrame(index=catda[time].values)
-    
+    df = pd.DataFrame(index=catda[tdim].values)
     for lat, lon in locations:
         indexer = np.column_stack([time_coord,
                                   [lat]*len(time_coord),
                                   [lon]*len(time_coord)])
-    
+
         series = f(indexer)
-        
         df[(lat,lon)] = series
-        
-    return(df)
-
+    return df
 
 if __name__ == '__main__':
-    interpolate_model_timeseries(locations=locations, times='all')
+    df = interpolate_model_timeseries(locations, tdim, times='all')
```

Comment:
Maybe this script should default to saving a csv if run from the command line (i.e. if main)